assessment/libs/complex.php

```diff
@@ -0,0 +1,636 @@
+    $tht=$thn+$thconj;
+    $re=$r*cos($tht);
+    $im=$r*sin($tht);  
+        
+    $st= array([round($re,$roundto) , round($im,$roundto)]);     
+        
+    return $st;
+}
+
+//-------------------------------------Quadratic real and complex roots--------------------------------------
+// Function: cx_quadRoot(a,b,c, [roundto = 12])
+// Returns an array of roots of the quadratic equation f(x) = ax^2 + bx + c. Real roots are returned
+// as an array([r1],[r2]) ordered from the smallest to largest, and the complex roots are returned 
+// as an array([Re1,Im1], [Re2,Im2]). 
+// 
+// Parameters:
+// a: The numerical coefficient of x^2
+// b: The numerical coefficient of x    
+// c: The constant
+// roundto: Optional - number of decimal places to which modulus should be rounded off; 
+//          default is 12 decimal places. 
+//
+// Returns:
+// An array of roots (either real or complex) of the quadratic equation.
+
+function cx_quadRoot(float $a, float $b, float $c, int $roundto = 12){
+    $d=$b**2 - 4*$a*$c;
+    if ($d<0){
+
+        $re= round(-$b/(2*$a),$roundto);
+        $im= round(sqrt(abs($d))/(2*$a),$roundto);
+        $im2=-$im;
+        $st=array([$re,$im],[$re,$im2]);
+    }
+        else {
+            $r1=round(((-$b-sqrt($d))/(2*$a)),$roundto);
+            $r2=round(((-$b+sqrt($d))/(2*$a)),$roundto);
+            $st=array([$r1],[$r2]);
+        }
+
+    return $st;
+
+}
+
+//------------------------------------------Formating to standard form-------------------------------------
+// Function: cx_format2std(num, [roundto=3])
+// Returns a string form of a complex number in standard form: a + b i. 
+// 
+// Parameters:
+// num: An array of real and imaginary parts of complex numbers given in square brackets: num = array([Re, Im]).
+//      num array can include more than one complex number. 
+// roundto: Optional - number of decimal places to which modulus should be rounded off; 
+//          default is 3 decimal places. 
+//
+// Returns:
+// A complex number in standard form as a string. If num has more than one complex number, the function returns
+// an array of the standard forms as strings: answer = array("a1+b1 i", "a2+b2 i", ...).
+
+function cx_format2std(array $num,int $roundto=3) {
+
+    $A=array();
+    for ($i=0;$i<count($num);$i++){
+        
+        $re=round($num[$i][0],$roundto);
+        $im=round($num[$i][1],$roundto);
+
+        $A[$i]=makexxpretty("$re + $im i");
+    }
+    
+    if (count($num)>1){
+        $ans=$A;
+    }    else{
+            $ans=$A[0];
+        }
+
+    return $ans;
+      
+}
+
+//------------------------------------------Formating to polar form---------------------------------------
+// Function: cx_format2pol(num, [argin="deg" , roundto=3])
+// Returns a string form of a complex number in polar form: r (cos(t) + isin(t)). 
+// 
+// Parameters:
+// num: An array of real and imaginary parts of complex numbers given in square brackets: num = array([Re, Im]).
```

Comment:
I'd suggest the same change as in the last function, to accept a single complex number or an array of complex numbers.